sycl/source/detail/xpti_registry.cpp

```diff
@@ -0,0 +1,100 @@
+//==---------- xpti_registry.cpp ----- XPTI Stream Registry ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <detail/global_handler.hpp>
+#include <detail/xpti_registry.hpp>
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+#include "xpti/xpti_trace_framework.hpp"
+#include <sstream>
+#endif
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+void XPTIRegistry::bufferConstructorNotification(
```

Comment:
We should probably rename it to `XPTISupport` or `XPTIHelper`...

Reply:
It can be done as  separate PR